server/grpc_service.go

```diff
@@ -1804,18 +1805,18 @@ func (s *GrpcServer) getTSOForwardStream(ctx context.Context, forwardedHost stri
 	}
 
 	// Now let's create the client connection and the forward stream
-	client, err := s.getDelegateClient(ctx, forwardedHost)
+	client, err := s.getDelegateClient(s.ctx, forwardedHost)
 	if err != nil {
 		return nil, err
 	}
 	done := make(chan struct{})
-	ctx, cancel := context.WithTimeout(s.ctx, tsoutil.DefaultTSOProxyTimeout)
+	ctx, cancel := context.WithCancel(s.ctx)
```

Comment:
From the code and your fix, it seems that #6254 has two issues: one is "context deadline exceeded", another is goroutine leak if TSOClient/tso-stream creation continuously returns error. 

It fixed the first issue by removing the tsoutil.DefaultTSOProxyTimeout, and I suspect this is the right fix. What's the reason to cause TSOClient/tso-stream creation to timeout? Shall we set larger timeout value instead of removing it?

Reply:
Just realized that checkStream has checked "case <-time.After(3 * time.Second):", that's why you removed tsoutil.DefaultTSOProxyTimeout here. checkStream() hard code the timeout value to 3 seconds, do we need to parameterize the timeout value and pass tsoutil.DefaultTSOProxyTimeout  to checkStream()?

Reply:
yes, you are right, `tsoutil.DefaultTSOProxyTimeout` is 3 seconds, too.